engine/runner/src/main/scala/org/enso/runner/Main.scala

```diff
@@ -843,11 +843,37 @@ object Main {
     */
   val defaultLogLevel: LogLevel = LogLevel.Error
 
+  var sampler : org.netbeans.modules.sampler.Sampler = org.netbeans.modules.sampler.Sampler.createSampler("LanguageServer")
+
+  def stopSampling(): Unit = {
+    if (sampler == null) {
+        return;
+    }
+    val where = "/tmp/language-server.npss"
+    System.err.println("Dumping out the sampled data " + where)
+    val out = new java.io.FileOutputStream(where)
+    val data = new java.io.DataOutputStream(out)
```

Comment:
Please use [scala.util.Using](https://www.scala-lang.org/api/2.13.8/scala/util/Using$.html) for automatic resource management